File: PostsAndComments/app/src/main/java/com/example/postscommentsapp/domain/usecase/InsertCommentUseCase.java
package com.example.postscommentsapp.domain.usecase;

import com.example.postscommentsapp.domain.model.Comment;
import com.example.postscommentsapp.domain.repository.CommentsRepository;

import javax.inject.Inject;

public class InsertCommentUseCase {

    private final CommentsRepository repository;

    @Inject
    public InsertCommentUseCase(CommentsRepository repository) {
        this.repository = repository;
    }

    public void execute(Comment comment) {
        repository.insertComment(comment);
    }
}

File: PostsAndComments/app/src/main/java/com/example/postscommentsapp/domain/usecase/InsertCommentUseCase.kt
package com.example.postscommentsapp.domain.usecase

import com.example.postscommentsapp.domain.model.Comment
import com.example.postscommentsapp.domain.repository.CommentsRepository
import javax.inject.Inject

class InsertCommentUseCase @Inject constructor(
    private val repository: CommentsRepository
) {
    suspend fun execute(comment: Comment) = repository.insertComment(comment)
}
